pkg/psmdb/mongo/mongo.go

```diff
@@ -32,11 +32,6 @@ func Dial(addrs []string, replset, username, password string, useTLS bool) (*mon
 		SetWriteConcern(writeconcern.New(writeconcern.WMajority(), writeconcern.J(true))).
 		SetReadPreference(readpref.Primary())
 
-	if useTLS {
-		tlsCfg := tls.Config{InsecureSkipVerify: true}
-		opts = opts.SetTLSConfig(&tlsCfg).SetDialer(tlsDialer{cfg: &tlsCfg})
-	}
-
```

Comment:
I believe if allowUnsafeConfiguration=false we should fetch CA from secret and use TLS connection between Operator and MongoDB